### PR TITLE
Feat:  Site Journey

### PR DESCRIPTION
**Pull request by byohannes**

### CODE REVIEW

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import Joi from 'joi'
+import pageRoute from '../../../routes/page-route.js'
+import { siteURIs } from '../../../uris.js'
+import { ukPostcodeRegex } from '../../contact/common/postcode/postcode-page.js'
+import { addressLookupForPostcode } from '../../contact/common/postcode/postcode.js'
+
+export const getData = async request => {
+  // await request.cache().clearPageData(siteURIs.SITE_GOT_POSTCODE.page)
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js:8]: Remove the comment please

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const { siteData } = (await request.cache().getData())
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js:9]: Why the ()?

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+export default pageRoute({
+  page: siteURIs.SITE_GOT_POSTCODE.page,
+  uri: siteURIs.SITE_GOT_POSTCODE.uri,
+  validator: Joi.object({
+    'site-postcode-check': Joi.any()
+      .valid('yes', 'no')
+      .required(),
+    'site-postcode': Joi.string().trim().min(1).max(12).required().pattern(ukPostcodeRegex)
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js:45]: I think the max UK postcode length is 7 characters (excluding the ' ') however I think the regex handles this so you can probably just remove the min, max and ensure the text entry is just long enough

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.njk [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    text: "Postcode"
+  }
+}) }}
+{% endset -%}
+{% set title = 'Does the site have a postcode?' %}
+{% block pageTitle %}
+  {{ 'Error: ' + title if error else title }}
+{% endblock %}
+{% block pageContent %}
+  {{ govukRadios({
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.njk:31]: Indentation

File: packages/web-service/src/pages/supporting-information/check-supporting-information.njk [hunk 5]
```diff
@@ -54,7 +54,7 @@
             errorMessage: { text: 'Select an option' } if error['another-file-check'],
             fieldset: {
               legend: {
-                text: "Do you want to upload another file?" if data.length > 0 else "Do you want to upload a supporting information file?" ,
+                text: "Do you want to upload another file ?" if data.length > 0 else "Do you want to upload a supporting information file ?" ,
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/supporting-information/check-supporting-information.njk:57]: Do you really want a space here?

**Review by GrahamMWillis — CHANGES_REQUESTED** [review]
Looks really good, just a couple of comments

File: packages/web-service/src/pages/site/site-check/site-check.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  if (pageData?.payload['address-and-grid-reference-mismatch'] === 'address') {
+    redirectUrl = siteURIs.SITE_GOT_POSTCODE.uri
+  } else if (pageData?.payload['address-and-grid-reference-mismatch'] === 'gridReference') {
+    redirectUrl = siteURIs.SITE_GRID_REF.uri
+  }
+
+  return redirectUrl
+}
+
+const siteAddressLine1 = c => [
```
[the remark this answers is not in the document]
Comment (byohannes) [reply packages/web-service/src/pages/site/site-check/site-check.js:20]: moved the address. to a service directory to be accessed commonly across other pages

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js (hunk 1 above)
Comment (byohannes) [reply packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js]: removed the commented code

File: packages/web-service/src/__tests__/additional-page-data.spec.js [hunk 7]
```diff
@@ -63,6 +63,11 @@ describe('additional page data', () => {
         methodExperience: '/enter-methods',
         previousLicence: '/previous-licence',
         register: '/register',
+        siteGridReference: '/site-grid-ref',
+        siteMap: '/upload-map',
+        siteMapThree: '/upload-map-3',
+        siteMapTwo: '/upload-map-2',
```
Comment (kealanparr) [inline packages/web-service/src/__tests__/additional-page-data.spec.js:69]: `upload-map-2`, `upload-map-2` - can we rename it to something more descriptive?

Comment (kealanparr) [reply packages/web-service/src/__tests__/additional-page-data.spec.js:69]: Please make sure the `CONFIRMED_NOT_COMPLETE` handler works too :) Just had a quick Slack call we need to go back to CYA page, if you complete the flow but don't submit the answers

**Review by kealanparr — DISMISSED** [review]
Comments left inline to please fix :)

**Review by GrahamMWillis — APPROVED** [review]

File: packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js (hunk 2 above)
Comment (byohannes) [reply packages/web-service/src/pages/site/site-got-postcode/site-got-postcode.js]: removed the brackets

**Review by kealanparr — APPROVED** [review]

**Review by kealanparr — APPROVED** [review]

**Pull request merged**